core/src/layers/prometheus.rs

```diff
@@ -760,3 +874,51 @@ impl<R: oio::BlockingWrite> oio::BlockingWrite for PrometheusMetricWrapper<R> {
         })
     }
 }
+
+fn get_path_label(path: &str, path_level: i32) -> String {
+    match path_level {
+        0 => "".to_string(),
+        -1 => path.to_string(),
+        n if n > 0 => {
```

Comment:
We can implement those logic like:

```rust
let idx: Option<usize> = s.char_indices()
        .filter(|&(_, c)| c == '/')
        .nth(n)
        .map(|(i, _)| i);
```

And can we return `&str` here to avoid an additional allocation?